### ISSUE TITLE
tycho-p2-extras-plugin::compare-version-with-baselines fails to compare repository with the same baseline

### ISSUE THREAD

**Issue by vrubezhny** [eclipse-tycho/tycho#1790]
The build of version 1.0.1  [WildWebDeveloperr project](https://github.com/eclipse/wildwebdeveloper) (which is the current latest release) fails on comparing of the `repository` project:

```
[ERROR] Failed to execute goal org.eclipse.tycho.extras:tycho-p2-extras-plugin:4.0.0-SNAPSHOT:compare-version-with-baselines (compare-version-with-baseline) on project repository: Baseline and reactor have same fully qualified version, but with different content
[ERROR] different
[ERROR]    Info.plist: present in baseline only
[ERROR]    MacOS/eclipse: present in baseline only
[ERROR]    artifacts.jar: not present in baseline
[ERROR]    artifacts.xml.xz: not present in baseline
[ERROR]    binary/org.eclipse.platform_root_4.25.0.v20220831-1800: not present in baseline
[ERROR]    binary/org.eclipse.rcp_root_4.25.0.v20220831-1800: not present in baseline
[ERROR]    binary/org.eclipse.wildwebdeveloper.product.executable.cocoa.macosx.aarch64_1.0.1.202211231523: not present in baseline
...
```
reporting a lot of artifacts as `not present in baseline`, while they are actually exist and of the same version (including the qualifiers) as the ones the  just built `repository` project consist of. The goal `compare-version-with-baselines` is configured to compare to the baseline repository at `http://download.eclipse.org/wildwebdeveloper/releases/latest` which currently contains the same version 1.0.1.

This behavior looks like a bug. 

Steps to reproduce:

- Clone the project, switch it to tag `1.0.1` and build skipping the tests:
```
$ git clone https://github.com/eclipse/wildwebdeveloper.git
$ cd wildwebdeveloper
$ git fetch origin
$ git checkout 1.0.1
$ mvn clean install -DskipTests
```
- Observe build errors

**Comment by laeubi** [comment]
> This behavior looks like a bug.

Can you provide an [integration-test](https://github.com/eclipse/tycho/wiki#providing-an-integration-test) to demonstrate the issue? I recently extracted the invoker test into an own class so it should be easier to enhance the test-cases now:

https://github.com/eclipse-tycho/tycho/blob/master/tycho-its/src/test/java/org/eclipse/tycho/test/P2ExtrasPlugin.java

@vrubezhny  do you plan a fix, as we are currently prepare for a 3.0.5 Tycho release this might then be worth being included.

**Comment by vrubezhny** [comment]
@laeubi No, I'm not planning to work on the issue, neither have a simple example project to be used in a test (using the whole WildWebDeveloper project for such a test looks like a bad idea). 
Also, unfortunately the version mentioned in the issue is too old and cannot be used anymore (it still cannot be built using that version, but the resulting errors are different and not related to this issue), as well as if the latest WWD version (1.1.0) is used the resulting error is also different due to the difference mostly in NPM dependency modules (they can be updated since the release date)
